gprofiler/profilers/perf.py

```diff
@@ -229,6 +237,11 @@ def _get_metadata(self, pid: int) -> Optional[AppMetadata]:
         return None
 
     def snapshot(self) -> ProcessToProfileData:
+        if self.perf_node_attach:
+            new_processes = [process for process in get_node_processes() if process not in self._node_processes]
+            generate_map_for_node_processes(new_processes)
+            self._node_processes.extend(new_processes)
```

Comment:
Prune dead processes, otherwise the list grows forever. I suppose the logic can be:
1. remove all processes where `not is_process_running()`
1. filter new from `get_node_processes()`
3. generate map for them
4. add to `_node_processes`